### ISSUE TITLE
MD034 is triggered when it is not supposed to

### ISSUE THREAD

**Issue by techmagus** [DavidAnson/vscode-markdownlint#235]
Hi,

There are some instances wherein MD034 is triggered when it is not supposed to (at least as per documentation).

Sample file:
Before the trigger: [access-google-ipv6.md](https://github.com/DavidAnson/vscode-markdownlint/files/8346538/access-google-ipv6.md)

Snippet:
```html
* Google Sites: <a href="https://sites.google.com" rel="noopener external nofollow">https://sites.google.com</a>
* Google Translate: <a href="https://translate.google.com" rel="noopener external nofollow">https://translate.google.com</a>
* Google Earth: <a href="https://www.google.com/earth/" rel="noopener external nofollow">https://earth.google.com</a>
* Google Moderator: <span class="popper animate removed_link" data-popper="https://moderator.appspot.com">https://web.archive.org/web/20150621104346/https://moderator.appspot.com/</span>
* Google FeedBurner: <a href="https://accounts.google.com/ServiceLogin?service=feedburner&amp;continue=https%3A%2F%2Ffeedburner.google.com%2Ffb%2Fa%2Fmyfeeds&amp;gsessionid=u9hHPZko-UNXqdtUCdPKIg" rel="noopener external nofollow">https://feedburner.google.com</a>
* Google YouTube: <a href="https://www.youtube.com/supported_browsers?next_url=%2F" rel="noopener external nofollow">https://youtube.com</a>
* Google App Engine: <a href="https://appengine.google.com" rel="noopener external nofollow">https://appengine.google.com</a>
* Google Code: <a href="https://code.google.com" rel="noopener external nofollow">https://code.google.com</a>
* Google Docs: <a href="https://docs.google.com" rel="noopener external nofollow">https://docs.google.com</a>
* Google Health: <a href="https://www.google.com/intl/en_us/health/about/" rel="noopener external nofollow">https://health.google.com</a>
```

aside: The md content was transferred from an old platform so it is not yet converted to markdown. Currently doing minor edits / passes when I noticed this bug(?).

After the trigger: [access-google-ipv6.md](https://github.com/DavidAnson/vscode-markdownlint/files/8346556/access-google-ipv6.md)

Snippet:
```html
- Google Sites: <a href="<https://sites.google.com>" rel="noopener external nofollow">https://sites.google.com</a>
- Google Translate: <a href="<https://translate.google.com>" rel="noopener external nofollow">https://translate.google.com</a>
- Google Earth: <a href="https://www.google.com/earth/" rel="noopener external nofollow"><https://earth.google.com></a>
- Google Moderator: <span class="popper animate removed_link" data-popper="https://moderator.appspot.com"><https://web.archive.org/web/20150621104346/https://moderator.appspot.com/></span>
- Google FeedBurner: <a href="https://accounts.google.com/ServiceLogin?service=feedburner&amp;continue=https%3A%2F%2Ffeedburner.google.com%2Ffb%2Fa%2Fmyfeeds&amp;gsessionid=u9hHPZko-UNXqdtUCdPKIg" rel="noopener external nofollow"><https://feedburner.google.com></a>
- Google YouTube: <a href="https://www.youtube.com/supported_browsers?next_url=%2F" rel="noopener external nofollow"><https://youtube.com></a>
- Google App Engine: <a href="<https://appengine.google.com>" rel="noopener external nofollow">https://appengine.google.com</a>
- Google Code: <a href="<https://code.google.com>" rel="noopener external nofollow">https://code.google.com</a>
- Google Docs: <a href="<https://docs.google.com>" rel="noopener external nofollow">https://docs.google.com</a>
- Google Health: <a href="https://www.google.com/intl/en_us/health/about/" rel="noopener external nofollow"><https://health.google.com></a>
```

Some triggers MD034 (`< >` are added) while some not.

Initially I thought it triggers when there is no trailing slash, however, there are URLs with a trailing slash that still gets edited by MD034.

Example:
```html
* Google Reader: <a href="https://www.google.com/reader/about/" rel="noopener external nofollow">https://www.google.com/reader/</a>
```

Is turned into:
```html
- Google Reader: <a href="<https://www.google.com/reader/>about/" rel="noopener external nofollow">https://www.google.com/reader/</a>
```

As a temporary workaround, I disabled MD034 but of course it means I won't get alerts about bare URLs.

My settings.json:
```json
{
  "auto-close-tag.SublimeText3Mode": true,
  "autoprefixer.formatOnSave": true,
  "editor.acceptSuggestionOnEnter": "smart",
  "editor.bracketPairColorization.enabled": true,
  "editor.codeActionsOnSave": {
    "source.fixAll.markdownlint": true
  },
  "editor.fontFamily": "'Noto Mono', 'Noto Sans Mono', 'monospace', monospace",
  "editor.linkedEditing": true,
  "editor.renderWhitespace": "all",
  "editor.rulers": [70, 80, 100, 120],
  "editor.showFoldingControls": "always",
  "editor.tabSize": 2,
  "editor.wordWrap": "on",
  "editor.wrappingIndent": "deepIndent",
  "explorer.autoReveal": false,
  "files.trimTrailingWhitespace": true,
  "html.autoClosingTags": false,
  "html.autoCreateQuotes": false,
  "markdownlint.config": {
    "default": true,
    "MD003": {
      "style": "atx"
    },
    "MD004": {
      "style": "dash"
    },
    "MD029": {
      "style": "one_or_ordered"
    },
    "MD033": {
      "allowed_elements": [
        "bdi",
        "bdo",
        "cite",
        "del",
        "ins",
        "mark",
        "q",
        "rb",
        "rbc",
        "rp",
        "rt",
        "ruby",
        "span",
        "sub",
        "sup"
      ]
    },
    "MD035": {
      "style": "---"
    },
    "MD046": {
      "style": "fenced"
    },
    "MD048": {
      "style": "backtick"
    },
    "MD049": {
      "style": "asterisk"
    },
    "MD050": {
      "style": "asterisk"
    }
  },
  "minify.minifyExistingOnSave": true,
  "redhat.telemetry.enabled": false,
  "todo-tree.general.enableFileWatcher": true,
  "todo-tree.highlights.useColourScheme": true,
  "window.zoomLevel": -0.5,
  "workbench.colorTheme": "Mayukai Semantic Mirage",
  "workbench.editor.closeEmptyGroups": false,
  "workbench.editor.enablePreview": false,
  "workbench.iconTheme": "material-icon-theme",
  "workbench.sideBar.location": "right"
}
```

Versions:
- VScode: 1.65.2
- markdownlint: 0.46.0

I hope I've provided all the necessary information.

Thank you. Shalom!
